appinventor/components/src/com/google/appinventor/components/runtime/ActivityStarter.java

```diff
@@ -359,6 +362,30 @@ public String ResultUri() {
     return "";
   }
 
+  /**
+   * Specifies the list of key-value pairs that will be passed as extra data to the activity.
+   */
+  @SimpleProperty
+  public void Extras(YailList pairs) {
```

Comment:
I don't think you've initialized Extras.   You should set it to an empty (Yail) list when the Activity Starter component is constructed.